app/models/ont/pool.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Ont
+  # Pool
+  class Pool < ApplicationRecord
+    belongs_to :tube, default: -> { Tube.new }
+
+    has_many :libraries, class_name: 'Ont::Library', foreign_key: :ont_pool_id,
+                         dependent: :destroy, inverse_of: :pool
+    has_many :requests, through: :libraries
+    # This is dependent on the requests association, so needs to be included
+    # after that is defined
+    include PlateSourcedLibrary
+    validates :volume, :concentration,
+              :insert_size, numericality: { greater_than_or_equal_to: 0, allow_nil: true }
+    validates :libraries, presence: true
+    validates_with TagValidator
+
+    before_save { self.final_library_amount = calculate_final_library_amount }
+
+    def library_attributes=(library_options)
```

Comment:
There are a few methods here which are exactly the same in Pacbio so is there an opportunity for sharing or is it too early?

Reply:
Yes I found something where you could assign an 'attribute' with a default value but I couldnt figure out a way to call any class methods like calculate_final_library_amount from the default assignment. Think it was an issue with the lambda scope but i couldnt figure it out.